### DISCUSSION TITLE
Expected behaviour with lr in vocoder training with restore path

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#250] [Archived]
<USER>>>> nmstoker<br/> <DATE> [August 3, 2020, 12:00pm]<br/> <br/> Most of my recent training run with the Vocoder was done with<br/> *slash* --continue_path but last night I switched to try fine-tuning with<br/> *slash* --restore_path and varying the LR. I was surprised as the LR seems to<br/> be following the original settings.<br/> <br/> **Q.** What's the expected behaviour with *slash* --restore_path?<br/> <br/> I'm using dev (72a6ac5). The lrD and lrG values it shows at the bottom<br/> of this quote are *not* what I've put in my config file, attached (which<br/> has 1e-4 now, although with the initial run one was 0.0001 and one was<br/> 1e-5, which was what had worked for me with the original external<br/> MelGAN). This seems to continue too (ie it's not just the first few<br/> evaluations that it's like that).<br/> <br/> Maybe it's some effect from the LR optimiser that I'm misunderstanding<br/> or not using correctly? I see below it mentions several 'Restoring...'<br/> lines which I believe indicate that it's using the original optimiser<br/> settings rather than reading from the config file even though I'm trying<br/> to fine-tune now.<br/> <br/> > CUDA_VISIBLE_DEVICES='0' python bin/train_vocoder.py *slash* --restore_path<br/> > .../.../models/neil17/multiband-melgan-July-27-2020_02+28PM-72a6ac5/checkpoint_1000000.pth.tar<br/> > *slash* --config_path<br/> > .../.../models/neil17/multiband-melgan-July-27-2020_02+28PM-72a6ac5/config.json<br/> > *slash* |& tee .../.../models/neil17/training_vocoder_n17_D11_02aug2020.log<br/> ><br/> > > Using CUDA: True *slash* <br/> > > Number of GPUs: 1 *slash* <br/> > > Git Hash: 72a6ac5 *slash* <br/> > > Experiment folder:<br/> > > /home/neil/main/Projects/TTSJul2020/models/neil17/multiband-melgan-August-02-2020_07+38PM-72a6ac5 *slash* <br/> > > Loading wavs from: /home/neil/data/Projects/NeilTTS/neil16/wavs/ *slash* <br/> > > Setting up Audio Processor... *slash* <br/> > > *slash* | *slash* > sample_rate:22050 *slash* <br/> > > *slash* | *slash* > num_mels:80 *slash* <br/> > > *slash* | *slash* > min_level_db:-10 *slash* <br/> > > *slash* | *slash* > frame_shift_ms:None *slash* <br/> > > *slash* | *slash* > frame_length_ms:None *slash* <br/> > > *slash* | *slash* > ref_level_db:0 *slash* <br/> > > *slash* | *slash* > fft_size:1024 *slash* <br/> > > *slash* | *slash* > power:None *slash* <br/> > > *slash* | *slash* > preemphasis:0.99 *slash* <br/> > > *slash* | *slash* > griffin_lim_iters:None *slash* <br/> > > *slash* | *slash* > signal_norm:True *slash* <br/> > > *slash* | *slash* > symmetric_norm:True *slash* <br/> > > *slash* | *slash* > mel_fmin:40.0 *slash* <br/> > > *slash* | *slash* > mel_fmax:8000.0 *slash* <br/> > > *slash* | *slash* > spec_gain:1.0 *slash* <br/> > > *slash* | *slash* > stft_pad_mode:reflect *slash* <br/> > > *slash* | *slash* > max_norm:4.0 *slash* <br/> > > *slash* | *slash* > clip_norm:True *slash* <br/> > > *slash* | *slash* > do_trim_silence:True *slash* <br/> > > *slash* | *slash* > trim_db:60 *slash* <br/> > > *slash* | *slash* > do_sound_norm:False *slash* <br/> > > *slash* | *slash* > stats_path:None *slash* <br/> > > *slash* | *slash* > hop_length:256 *slash* <br/> > > *slash* | *slash* > win_length:1024 *slash* <br/> > > Generator Model: multiband_melgan_generator *slash* <br/> > > Discriminator Model: melgan_multiscale_discriminator *slash* <br/> > > Restoring Generator Model... *slash* <br/> > > Restoring Generator Optimizer... *slash* <br/> > > Restoring Discriminator Model... *slash* <br/> > > Restoring Discriminator Optimizer... *slash* <br/> > > Restoring Generator LR Scheduler... *slash* <br/> > > Restoring Discriminator LR Scheduler... *slash* <br/> > > Model restored from step 1000000 *slash* <br/> > > Generator has 2539448 parameters *slash* <br/> > > Discriminator has 4354998 parameters<br/> ><br/> > > EPOCH: 0/10000<br/> ><br/> > > TRAINING (2020-08-02 19:38:21)<br/> ><br/> > -- *slash* > STEP: 199/493 -- GLOBAL_STEP: 1000200 *slash* <br/> > *slash* | *slash* > G_stft_loss_mg: 0.77097 (0.77146) *slash* <br/> > *slash* | *slash* > G_stft_loss_sc: 0.48925 (0.51256) *slash* <br/> > *slash* | *slash* > G_subband_stft_loss_mg: 0.64009 (0.64357) *slash* <br/> > *slash* | *slash* > G_subband_stft_loss_sc: 0.49436 (0.52802) *slash* <br/> > *slash* | *slash* > G_mse_fake_loss: 0.51116 (0.42568) *slash* <br/> > *slash* | *slash* > G_loss: 2.47523 (2.29200) *slash* <br/> > *slash* | *slash* > G_gen_loss: 1.19733 (1.22781) *slash* <br/> > *slash* | *slash* > G_adv_loss: 1.27790 (1.06419) *slash* <br/> > *slash* | *slash* > D_mse_gan_loss: 0.36095 (0.38322) *slash* <br/> > *slash* | *slash* > D_mse_gan_real_loss: 0.15346 (0.13951) *slash* <br/> > *slash* | *slash* > D_mse_gan_fake_loss: 0.11444 (0.12729) *slash* <br/> > *slash* | *slash* > D_loss: 0.36095 (0.38322) *slash* <br/> > *slash* | *slash* > step_time: 0.68 *slash* <br/> > *slash* | *slash* > loader_time: 0.00 *slash* <br/> > *slash* | *slash* > lrG: 0.0001 *slash* <br/> > *slash* | *slash* > lrD: 0.0001<br/> <br/> [config.zip](https://discourse.mozilla.org/uploads/short-url/6tvrBFtYUFzOuGmqNvIMw1KNZck.zip)<br/> (2.8 KB)<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/expected-behaviour-with-lr-in-vocoder-training-with-restore-path]<br/>

**Comment by JRMeyer** [comment]
## [Archived] Expected behaviour with lr in vocoder training with restore path<br/> <br/> <USER>>>> nmstoker<br/> <DATE> [August 3, 2020, 12:00pm]<br/> <br/> Most of my recent training run with the Vocoder was done with<br/> *slash* --continue_path but last night I switched to try fine-tuning with<br/> *slash* --restore_path and varying the LR. I was surprised as the LR seems to<br/> be following the original settings.<br/> <br/> **Q.** What's the expected behaviour with *slash* --restore_path?<br/> <br/> I'm using dev (72a6ac5). The lrD and lrG values it shows at the bottom<br/> of this quote are *not* what I've put in my config file, attached (which<br/> has 1e-4 now, although with the initial run one was 0.0001 and one was<br/> 1e-5, which was what had worked for me with the original external<br/> MelGAN). This seems to continue too (ie it's not just the first few<br/> evaluations that it's like that).<br/> <br/> Maybe it's some effect from the LR optimiser that I'm misunderstanding<br/> or not using correctly? I see below it mentions several 'Restoring...'<br/> lines which I believe indicate that it's using the original optimiser<br/> settings rather than reading from the config file even though I'm trying<br/> to fine-tune now.<br/> <br/> > CUDA_VISIBLE_DEVICES='0' python bin/train_vocoder.py *slash* --restore_path<br/> > .../.../models/neil17/multiband-melgan-July-27-2020_02+28PM-72a6ac5/checkpoint_1000000.pth.tar<br/> > *slash* --config_path<br/> > .../.../models/neil17/multiband-melgan-July-27-2020_02+28PM-72a6ac5/config.json<br/> > *slash* |& tee .../.../models/neil17/training_vocoder_n17_D11_02aug2020.log<br/> ><br/> > > Using CUDA: True *slash* <br/> > > Number of GPUs: 1 *slash* <br/> > > Git Hash: 72a6ac5 *slash* <br/> > > Experiment folder:<br/> > > /home/neil/main/Projects/TTSJul2020/models/neil17/multiband-melgan-August-02-2020_07+38PM-72a6ac5 *slash* <br/> > > Loading wavs from: /home/neil/data/Projects/NeilTTS/neil16/wavs/ *slash* <br/> > > Setting up Audio Processor... *slash* <br/> > > *slash* | *slash* > sample_rate:22050 *slash* <br/> > > *slash* | *slash* > num_mels:80 *slash* <br/> > > *slash* | *slash* > min_level_db:-10 *slash* <br/> > > *slash* | *slash* > frame_shift_ms:None *slash* <br/> > > *slash* | *slash* > frame_length_ms:None *slash* <br/> > > *slash* | *slash* > ref_level_db:0 *slash* <br/> > > *slash* | *slash* > fft_size:1024 *slash* <br/> > > *slash* | *slash* > power:None *slash* <br/> > > *slash* | *slash* > preemphasis:0.99 *slash* <br/> > > *slash* | *slash* > griffin_lim_iters:None *slash* <br/> > > *slash* | *slash* > signal_norm:True *slash* <br/> > > *slash* | *slash* > symmetric_norm:True *slash* <br/> > > *slash* | *slash* > mel_fmin:40.0 *slash* <br/> > > *slash* | *slash* > mel_fmax:8000.0 *slash* <br/> > > *slash* | *slash* > spec_gain:1.0 *slash* <br/> > > *slash* | *slash* > stft_pad_mode:reflect *slash* <br/> > > *slash* | *slash* > max_norm:4.0 *slash* <br/> > > *slash* | *slash* > clip_norm:True *slash* <br/> > > *slash* | *slash* > do_trim_silence:True *slash* <br/> > > *slash* | *slash* > trim_db:60 *slash* <br/> > > *slash* | *slash* > do_sound_norm:False *slash* <br/> > > *slash* | *slash* > stats_path:None *slash* <br/> > > *slash* | *slash* > hop_length:256 *slash* <br/> > > *slash* | *slash* > win_length:1024 *slash* <br/> > > Generator Model: multiband_melgan_generator *slash* <br/> > > Discriminator Model: melgan_multiscale_discriminator *slash* <br/> > > Restoring Generator Model... *slash* <br/> > > Restoring Generator Optimizer... *slash* <br/> > > Restoring Discriminator Model... *slash* <br/> > > Restoring Discriminator Optimizer... *slash* <br/> > > Restoring Generator LR Scheduler... *slash* <br/> > > Restoring Discriminator LR Scheduler... *slash* <br/> > > Model restored from step 1000000 *slash* <br/> > > Generator has 2539448 parameters *slash* <br/> > > Discriminator has 4354998 parameters<br/> ><br/> > > EPOCH: 0/10000<br/> ><br/> > > TRAINING (2020-08-02 19:38:21)<br/> ><br/> > -- *slash* > STEP: 199/493 -- GLOBAL_STEP: 1000200 *slash* <br/> > *slash* | *slash* > G_stft_loss_mg: 0.77097 (0.77146) *slash* <br/> > *slash* | *slash* > G_stft_loss_sc: 0.48925 (0.51256) *slash* <br/> > *slash* | *slash* > G_subband_stft_loss_mg: 0.64009 (0.64357) *slash* <br/> > *slash* | *slash* > G_subband_stft_loss_sc: 0.49436 (0.52802) *slash* <br/> > *slash* | *slash* > G_mse_fake_loss: 0.51116 (0.42568) *slash* <br/> > *slash* | *slash* > G_loss: 2.47523 (2.29200) *slash* <br/> > *slash* | *slash* > G_gen_loss: 1.19733 (1.22781) *slash* <br/> > *slash* | *slash* > G_adv_loss: 1.27790 (1.06419) *slash* <br/> > *slash* | *slash* > D_mse_gan_loss: 0.36095 (0.38322) *slash* <br/> > *slash* | *slash* > D_mse_gan_real_loss: 0.15346 (0.13951) *slash* <br/> > *slash* | *slash* > D_mse_gan_fake_loss: 0.11444 (0.12729) *slash* <br/> > *slash* | *slash* > D_loss: 0.36095 (0.38322) *slash* <br/> > *slash* | *slash* > step_time: 0.68 *slash* <br/> > *slash* | *slash* > loader_time: 0.00 *slash* <br/> > *slash* | *slash* > lrG: 0.0001 *slash* <br/> > *slash* | *slash* > lrD: 0.0001<br/> <br/> [config.zip](https://discourse.mozilla.org/uploads/short-url/6tvrBFtYUFzOuGmqNvIMw1KNZck.zip)<br/> (2.8 KB)<br/> <br/> ### This is an archived TTS discussion thread from discourse.mozilla.org/t/expected-behaviour-with-lr-in-vocoder-training-with-restore-path<br/>
